spec.html

```diff
@@ -266,6 +266,47 @@ <h1>NotifyWaiter ( _WL_, <del>_W_</del><ins>_waiterRecord_</ins> )</h1>
     </emu-note>
   </emu-clause>
 
+  <emu-clause id="sec-dowait" aoid="DoWait">
+    <h1>DoWait ( _mode_, _typedArray_, _index_, _value_, _timeout_ )</h1>
+    <p><ins>This is a new abstract operation.</ins></p>
+    <p>The abstract operation DoWait takes five arguments, a String _mode_, and values _typedArray_, _idnex_, _value_, and _timeout_. It performs the following steps:</p>
+    <emu-alg>
+      1. Assert: _mode_ is either `"sync"` or `"async"`.
+      1. Let _buffer_ be ? ValidateSharedIntegerTypedArray(_typedArray_, *true*).
+      1. Let _i_ be ? ValidateAtomicAccess(_typedArray_, _index_).
+      1. Let _v_ be ? ToInt32(_value_).
+      1. Let _q_ be ? ToNumber(_timeout_).
+      1. If _q_ is *NaN*, let _t_ be *+&infin;*, else let _t_ be max(_q_, 0).
+      1. If _mode_ is `"sync"`, then
+        1. Let _B_ be AgentCanSuspend().
+        1. If _B_ is *false*, throw a *TypeError* exception.
```

Comment:
can you elaborate on why async doesn't need to check "AgentCanSuspend", but sync does?

Reply:
It's the whole point of `waitAsync`: `wait` blocks the thread and suspends it until it's notified. There are some threads that cannot suspend at all, like the browser main thread, thus the async version being proposed here.

Due to copypasta the `AgentCanSuspend()` check was still being checked by `Atomics.waitAsync`, this corrects it.